lib/Runtime/Library/RegexHelper.cpp

```diff
@@ -665,24 +665,57 @@ namespace Js
         CharCount inputLength = input->GetLength();
         UnifiedRegex::GroupInfo match; // initially undefined
 
+        const bool isGlobal = pattern->IsGlobal();
+        const bool isSticky = pattern->IsSticky();
+        const bool useCache = !isGlobal && !isSticky;
+
+        RegExpTestCache* cache = nullptr;
+        JavascriptString * cachedValue = nullptr;
+        uint cacheIndex = 0;
+        if (useCache)
+        {
+            cache = regularExpression->EnsureTestCache();
+            cacheIndex = JavascriptRegExp::GetTestCacheIndex(input);
+            cachedValue = cache[cacheIndex].input;
+        }
+
 #if ENABLE_REGEX_CONFIG_OPTIONS
         RegexHelperTrace(scriptContext, UnifiedRegex::RegexStats::Test, regularExpression, input);
+        JavascriptRegExp::TraceTestCache(cachedValue == input, input, cachedValue, !useCache);
 #endif
-        const bool isGlobal = pattern->IsGlobal();
-        const bool isSticky = pattern->IsSticky();
+
+        if (useCache && cachedValue == input)
+        {
+            return JavascriptBoolean::ToVar(cache[cacheIndex].result, scriptContext);
```

Comment:
>return JavascriptBoolean::ToVar(cache[cacheIndex].result, scriptContext); [](start = 12, length = 73)

In case of a cache hit Is it still worth to do the slow lookup in chk build and validate against cached value?

Reply:
seems like a good idea